pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -118,26 +122,49 @@ def advance(self, *args: Any, **kwargs: Any) -> None:
             # indicate the loop has run
             self._has_run = True
 
+    def on_advance_end(self) -> None:
+        # inform logger the epoch loop has finished
+        self.trainer.logger_connector.epoch_end_reached()
+
+        self._logged_outputs.append(self.trainer.logger_connector.update_eval_epoch_metrics())
```

Comment:
Now, these are computed when each dataloader ends, instead of when all dataloaders end

Reply:
quick ques: where are taking care of the condition that this method will only compute metrics for the current dataloader_idx?

Reply:
This will call `ResultCollection.metrics(False)` which in turn calls `ResultCollection.valid_items()` where we filter with the `ResultCollection.dataloader_idx` value

The LoggerConnector makes sure the `ResultCollection.dataloader_idx` value is the correct one

Reply:
> why is there so much space between lines? minimal distance rule for covid does not apply for code.

It's the general style used int he rest of this file. I also prefer tighter code. Feel free to push whitespace changes to your liking!

Reply:
Also, I think this might create some metric problems when doing cross-dataloaders aggregation as the first compute would be done on the first datalaoder and _compute would be stored.

Here is the failing test:

```py
def test_multiple_dataloaders_logging(tmpdir):

    class TestModel(BoringModel):

        def validation_step(self, batch, batch_idx, dataloader_idx):
            self.log("value_1", dataloader_idx, add_dataloader_idx=False)
            self.log("value_2", dataloader_idx, add_dataloader_idx=True)

        def val_dataloader(self):
            return [self.train_dataloader(), self.train_dataloader()]

    model = TestModel()
    model.validation_epoch_end = None
    trainer = Trainer(default_root_dir=tmpdir)
    results = trainer.validate(model)
    assert results == [{'value_2/dataloader_idx_0': 0.0, 'value_1': 0.5}, {'value_2/dataloader_idx_1': 1.0, 'value_1': 0.5}]
```

Reply:
Here is a PR to discuss the right behaviour: https://github.com/PyTorchLightning/pytorch-lightning/pull/11125/files

Reply:
This would need better investigation, but it seems this is confusing. I will open an issue and unblock this PR.